lib/src/views/Calendar/Day.tsx

```diff
@@ -251,6 +251,7 @@ export const areDayPropsEqual = (prevProps: DayProps, nextProps: DayProps) => {
     prevProps.showDaysOutsideCurrentMonth === nextProps.showDaysOutsideCurrentMonth &&
     prevProps.disableHighlightToday === nextProps.disableHighlightToday &&
     prevProps.className === nextProps.className &&
+    prevProps.inCurrentMonth === nextProps.inCurrentMonth &&
```

Comment:
> Didn't reproduce without reduceAnimation because transition group is mounting and unmounting the whole component on animation.

What do you think of dropping the `areDayPropsEqual` logic? I believe that if we drop the ButtonBase component as proposed #1674, we would make this optimization with almost no ROI.

Reply:
We could test it, but without this function everything is SO slow especially for dateRange picker.

Reply:
I'm confused. Here are my thoughts:

- What's the difference with the date range picker?
- We display two calendars.
- Ok so this is only an issue with desktop.
- On desktop we don't use `reduceAnimation`.
- You said that the logic is only working if `reduceAnimation`

I come to the conclusion that it shouldn't have any impact on date range picker. Where is the error?

Reply:
No it also works for DateRangePicker preview and in month selection and keyboard control.

Reply:
> keyboard control

The focus keyboad logic could be bypassing React entirely